common-ktx/src/main/java/co/nimblehq/common/extensions/DateExt.kt

```diff
@@ -0,0 +1,315 @@
+        calendar.time = this
+        return calendar.get(Calendar.DAY_OF_WEEK) == Calendar.SUNDAY
+    }
+
+/**
+ * Check if the current date is Monday
+ */
+val Date.isMonday: Boolean
+    get() {
+        val calendar = Calendar.getInstance()
+        calendar.time = this
+        return calendar.get(Calendar.DAY_OF_WEEK) == Calendar.MONDAY
+    }
+
+/**
+ * Check if the current date is Tuesday
+ */
+val Date.isTuesday: Boolean
+    get() {
+        val calendar = Calendar.getInstance()
+        calendar.time = this
+        return calendar.get(Calendar.DAY_OF_WEEK) == Calendar.TUESDAY
+    }
+
+/**
+ * Check if the current date is Wednesday
+ */
+val Date.isWednesday: Boolean
+    get() {
+        val calendar = Calendar.getInstance()
+        calendar.time = this
+        return calendar.get(Calendar.DAY_OF_WEEK) == Calendar.WEDNESDAY
+    }
+
+/**
+ * Check if the current date is Thursday
+ */
+val Date.isThursday: Boolean
+    get() {
+        val calendar = Calendar.getInstance()
+        calendar.time = this
+        return calendar.get(Calendar.DAY_OF_WEEK) == Calendar.THURSDAY
+    }
+
+/**
+ * Check if the current date is Friday
+ */
+val Date.isFriday: Boolean
+    get() {
+        val calendar = Calendar.getInstance()
+        calendar.time = this
+        return calendar.get(Calendar.DAY_OF_WEEK) == Calendar.FRIDAY
+    }
+
+/**
+ * Check if the current date is Saturday
+ */
+val Date.isSaturday: Boolean
+    get() {
+        val calendar = Calendar.getInstance()
+        calendar.time = this
+        return calendar.get(Calendar.DAY_OF_WEEK) == Calendar.SATURDAY
+    }
+
+/**
+ * Check if the current date is Weekend
+ */
+val Date.isWeekend: Boolean
+    get() = this.isSaturday || this.isSunday
+
+/**
+ * Check if the current date is Weekday
+ */
+val Date.isWeekday: Boolean
+    get() = !this.isWeekend
+
+/**
+ * Check if the current date is Today
+ */
+val Date.isToday: Boolean
+    get() = isDateIn(this, 0)
+
+/**
+ * Check if the current date is Yesterday
+ */
+val Date.isYesterday: Boolean
+    get() = isDateIn(this, -1)
+
+/**
+ * Check if the current date is Tomorrow
+ */
+val Date.isTomorrow: Boolean
+    get() = isDateIn(this, 1)
+
+/**
+ * Check if the date is in a specific time
+ *
+ * @param date - Date want to check
+ * @param variable - time of date
+ *
+ * @return value to check if the current date is in expected date
+ */
+private fun isDateIn(date: Date, variable: Int = 0): Boolean {
+    val now = Calendar.getInstance()
+    val cdate = Calendar.getInstance()
+    cdate.timeInMillis = date.time
+
+    now.add(Calendar.DATE, variable)
+
+    return (now.get(Calendar.YEAR) == cdate.get(Calendar.YEAR)
+        && now.get(Calendar.MONTH) == cdate.get(Calendar.MONTH)
+        && now.get(Calendar.DATE) == cdate.get(Calendar.DATE))
+}
+
+/**
+ * Convert from String to Date
+ */
+fun String.toDate(format: String): Date? = try{
+    SimpleDateFormat(format).parse(this)
+}catch (e : Exception) {
```

Comment:
🔍 We need a space here 😄

Reply:
@hoangnguyen92dn I don't thing it's a good idea 😄

Reply:
```suggestion
} catch (e : Exception) {
```
@doannimble Why not? 😄

Reply:
@lydiasama @hoangnguyen92dn Oops, I thought it was a line break, let me do the format code 🙏